cupy/_core/fusion.pyx

```diff
@@ -119,7 +119,15 @@ class _FusionVarCUDA(object):
         elif isinstance(val, complex):
             init = '({}, {})'.format(c.real, c.imag)
         elif isinstance(val, (int, float)):
-            init = '= {}'.format(c)
+            if numpy.isnan(val):
+                init = '= __int_as_float(0x7f800001)'
```

Comment:
nit:
/usr/local/cuda/include/math_constants.h
seems to define `CUDART_NAN` `CUDART_NAN_F` and INF ones
This is purely a cosmetic change as these values will not change, but you can use the mnemonic if you feel in the mood :)

Reply:
Fixed!